lychee-lib/src/collector.rs

```diff
@@ -72,13 +72,10 @@ impl Collector {
         // Filter out already cached links (duplicates)
         links.retain(|l| !self.cache.contains(&l.uri));
 
-        self.update_cache(&links);
-        Ok(links)
-    }
-
-    /// Update internal link cache
-    fn update_cache(&mut self, links: &HashSet<Request>) {
+        // Add remaining new links to cache
         self.cache.extend(links.iter().cloned().map(|l| l.uri));
```

Comment:
I reverted the previous attempts to fix it and simplified the code slightly by removing the dedicated function. Reading cache and filling cache are done only here right after another, so it should be fine and clean to keep it like that.

I'm still puzzled how the (changed) test can fail: Two tiny files are parsed, which spawn two `collect_links` instances (?), and both finish `links.retain` the exact same time before doing `self.cache.extend`? So we have a change to merge the `links.retain` into the `links.iter()`, to do this per URI, either adding it to the cache, if not present yet, or removing that element?